
# Acquiring the Reads

*B. napus* Illumina sequencing data was acquired from a previous study by de Jong and Adams (2023), [available on NCBI's SRA Run Selector](https://www.ncbi.nlm.nih.gov/Traces/study/?acc=GSE201225&f=genotype_sam_ss%3An%3Anatural%2520b.%2520napus%3Ac&o=acc_s%3Aa&s=SRR18856171,SRR18856174,SRR18856181,SRR18856184,SRR18856187,SRR18856194); only sequencing data from natural polyploids were selected, for a total of six `.sra` files. An accession list in `.txt` format (`SRR_Acc_List.txt`) was uploaded by the `scp` command to DRAC's Graham cluster, then used to fetch the `.sra` files by the following command:

`prefetch --option-file ./accession_list/SRR_Acc_List.txt`

The `.sra` read files were converted to `.fastq` format by the command:

`fasterq-dump --split-files SRR*/*`

And FASTQC was performed with:

`fastqc [SRR_dir_name]/*fastq*` where each of the six directory names (each corresponding to a pair of `.fastq` files, due to the accessions representing paired-end sequencing data).

Note that `fastqc SRR*/*.fastq` was tried previously, but only completed two accessions/`.fastq` pairs before aborting.

# Acquiring the Reference

The *B. napus* reference genome generated by Bayer et al. (2021) was downloaded from [this University of Western Australia page page](https://research-repository.uwa.edu.au/en/datasets/brassica-oleracea-rapa-napus-pangenome-data) -linked in the [authors' PubMed publication](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC8633514/) for the paper- to the Graham cluster by the command:

`wget https://research-repository.uwa.edu.au/files/87864413/Darmor_NRGene_pseudomolecules_v1.4_chloroplast_mitochondrium_both_rounds.renamed.fasta`

and a `md5sum` check returned a hash of `eb8bed1671b82b2a645ba554cd11d281`.

An annotation file was downloaded by the command:

`wget https://research-repository.uwa.edu.au/files/87864406/Darmor_NRGene_pseudomolecules_v1.4_chloroplast_mitochondrium_both_rounds_annotation.sorted.BrassicaNamingStandard.fixed.gff3.gz`

and its `md5sum` hash was `a7e5c2b8dde38ab909ecc9f91f4f50ca`. The annotation file was then unzipped with `gunzip`.

**Note:** Due to the planned scope of the project, a pangenome should be downloaded too. However, the student has yet to fully understand what the different file names given by Bayer et al. (2021) refer to.

# Generating the Index (ERROR THROWN)

Indexing of the reference genome was attempted with the following command:

`STAR --runMode genomeGenerate --genomeDir ./star_index --genomeFastaFiles ./ref_genome/Darmor_NRGene_pseudomulecules_v1.4_chloroplast_mitochondrium_both_rounds.renamed.fasta --sjdbGETfile ./ref_genome/Darmor_NRGene_pseudomolecules_v1.4_chloroplast_mitochondrium_both_rounds_annotation.sorted.BrassicaNamingStandard.fixed.gff3 --runThreadN 10 --sjdbGTFtagExonParentTranscript Parent --limitGenomeGenerateRAM 200000000000`

Which terminated after throwing an instance of `std::bad_alloc`. A previous run with `runThreadN 20` and no specified argument for `limitGenomeGenerateRAM` threw a fatal error and a prompt as follows:

`EXITING because of FATAL PARAMETER ERROR: limitGenomeGenerateRAM=310000000000 is too small for your genome. SOLUTION: please specify --limitGenomeGenerateRAM not less than 116287777717 and make that much RAM available.`